File: hmi/sdk/Linux/cpp/src/api/inspection.cpp
// Copyright (c) 2026 WeGo Robotics. All rights reserved.

#include "api/detail.hpp"
#include "shalom/api.hpp"

namespace shalom {

std::string RobotApi::triggerCapture(const std::string &vehicleNumber, const std::string &carNumber,
                                     const std::string &pointId, std::optional<int> tagId,
                                     std::string *err)
{
    std::string json = "{\"vehicle_number\":\"" + api::detail::escape(vehicleNumber)
                       + "\",\"car_number\":\"" + api::detail::escape(carNumber)
                       + "\",\"point_id\":\"" + api::detail::escape(pointId) + "\"";
    if (tagId)
        json += ",\"tag_id\":" + std::to_string(*tagId);
    return client_.sendRequest("cmd/capture/trigger", json + '}', err);
}

std::string RobotApi::armPreset(const std::string &name, std::string *err)
{
    if (name.empty())
        return api::detail::invalid(err, "arm preset must not be empty");
    return client_.sendRequest("cmd/arm/preset", "{\"name\":\"" + api::detail::escape(name) + "\"}", err);
}

std::string RobotApi::armJointGoal(const std::vector<double> &positions, std::string *err)
{
    if (positions.empty())
        return api::detail::invalid(err, "arm joint goal must not be empty");
    std::string values = "[";
    for (std::size_t i = 0; i < positions.size(); ++i) {
        if (!api::detail::finite(positions[i]))
            return api::detail::invalid(err, "arm joint goal must contain finite numbers");
        if (i)
            values += ',';
        values += api::detail::number(positions[i]);
    }
    return client_.sendRequest("cmd/arm/joint_goal", "{\"positions\":" + values + "]}", err);
}

std::string RobotApi::armStop(std::string *err)
{
    return client_.sendRequest("cmd/arm/stop", "{}", err);
}

}  // namespace shalom

File: hmi/sdk/Linux/python/shalom_sdk/api/inspection.py
"""Capture and commissioning-only arm commands."""

from __future__ import annotations

from typing import Any, Optional, Sequence

from ..types import Response
from ._base import _ApiBase


class InspectionApi(_ApiBase):
    def trigger_capture(self, vehicle_number: str = "UNKNOWN", car_number: str = "00",
                        point_id: str = "MANUAL", tag_id: Optional[int] = None,
                        timeout_s: float = 3.0) -> Response:
        payload: dict[str, Any] = {"vehicle_number": vehicle_number, "car_number": car_number,
                                   "point_id": point_id}
        if tag_id is not None:
            payload["tag_id"] = tag_id
        return self._client.request("cmd/capture/trigger", payload, timeout_s)

    # Do not expose arm methods in an operator UI until the delivered FR3
    # authority path has been approved.
    def arm_preset(self, name: str, timeout_s: float = 3.0) -> Response:
        if not name:
            raise ValueError("arm preset must not be empty")
        return self._client.request("cmd/arm/preset", {"name": name}, timeout_s)

    def arm_joint_goal(self, positions: Sequence[float], timeout_s: float = 3.0) -> Response:
        values = list(positions)
        if not values:
            raise ValueError("arm joint goal must not be empty")
        self._finite(*values)
        return self._client.request("cmd/arm/joint_goal", {"positions": values}, timeout_s)

    def arm_stop(self, timeout_s: float = 3.0) -> Response:
        return self._client.request("cmd/arm/stop", timeout_s=timeout_s)
